rules-reviewed/quarkus1/quarkus1-10/quarkus1-11-mongodb-panache.windup.xml

```diff
@@ -0,0 +1,69 @@
+    <metadata>
+        <description>
+            This ruleset gives hints for upgrading to quarkus 1.11 for applications that use MongoDB Panache
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-xml,3.0.0.Final" />
+        </dependencies>
+        <!-- The major version is implied by the target, the minor is reflected in the verionRange -->
+        <sourceTechnology id="quarkus1" versionRange="(,10]"/>
+        <targetTechnology id="quarkus1" versionRange="[11,)"/>
+    </metadata>
+    <rules>
+        <rule id="quarkus1-11-mongodb-panache-00000">
+            <when>
+                <graph-query discriminator="WindupJavaConfigurationModel" as="discard">
+                    <property type="BOOLEAN" name="sourceMode">true</property>
+                </graph-query>
+                <project>
+                    <artifact groupId="io.quarkus" artifactId="quarkus-mongodb-panache"/>
+                </project>
+            </when>
+            <perform>
+                <iteration over="default">
+                    <hint title="MongoDB Panache changed to conform with Hibernate Panache behavior" effort="1" category-id="potential">
+                        <message>
+                            A recent change was made to MongoDB panache to bring it in to conformity with the Hibernate Panache behavior. 
+                            Public field accesses on `MongoEntity` and `ReactiveMongoEntity` types are now wrapped with the appropriate `get` or `set` methods. 
+                            In general, you will like not see any difference in your application. 
+                            However, if you have written a custom `get` or `set` method you may notice a change in behavior if those custom methods deviate from the standard `get`/`set` paradigm. 
+                        </message>
+                        <link title="Quarkus - Simplified MongoDB with Panache" href="https://quarkus.io/guides/mongodb-panache" />
+                        <link title="Quarkus - Migraton Guide 1.11" href="https://github.com/quarkusio/quarkus/wiki/Migration-Guide-1.11" />
+                    </hint>
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="quarkus1-11-mongodb-panache-00010">
+            <when>
+                <or>
+                    <javaclass references="io.quarkus.mongodb.panache.PanacheMongoEntity">
+                        <location>INHERITANCE</location>
+                    </javaclass>
+                    <javaclass references="io.quarkus.mongodb.panache.PanacheMongoEntityBase">
+                        <location>INHERITANCE</location>
+                    </javaclass>
```

Comment:
if user doesnt write its own getter/setter , then will be no problem, according to doc, so why raising this issue always.

Reply:
As above Jonathan - it's an awareness thing and getting the balance right in terms of practicality.